utilities/transactions/write_prepared_txn.cc

```diff
@@ -23,33 +23,51 @@
 #include "util/cast_util.h"
 #include "utilities/transactions/pessimistic_transaction.h"
 #include "utilities/transactions/write_prepared_txn_db.h"
+#include "utilities/transactions/write_unprepared_txn.h"
 
 namespace rocksdb {
 
 struct WriteOptions;
 
+bool WritePreparedTxnReadCallback::IsCommitted(SequenceNumber seq) {
+  if (txn_ != nullptr) {
+    auto unprep_seqs = txn_->GetUnpreparedSequenceNumbers();
```

Comment:
I agree the naming is a bit confusing, since I'm using WritePreparedTxnReadCallback for both write prepared and unprepared, but I can change that. This is so that I can have it embedded this class on the WritePreparedTxn object.

If I make a separate class, I'd waste slightly more memory, but it'll be less confusing and slightly more efficient for write prepared. Any preference on this?

Reply:
by embedding it you mean making it on-stack in WritePreparedTxn::Get? If yes then I guess we can override WritePreparedTxn::Get in WriteUnpreparedTxn? The function is small anyway.